File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        hl , nl = len(haystack) , len(needle)
        i = 0

        if nl > hl:
            return -1 

        while i+nl <= hl:
            if haystack[i:i+nl] == needle:
                return i
            i+=1
        return -1

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.go
func strStr(haystack string, needle string) int {
    hl , nl := len(haystack) , len(needle)

    if nl > hl {
        return -1
    }

    i := 0
    for i + nl <= hl {
        if haystack[i:i+nl] == needle{
            return i
        }
        i++
    }

    return -1

}